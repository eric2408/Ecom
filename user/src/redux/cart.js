import {createSlice} from '@reduxjs/toolkit';


const cart = createSlice({
    name:"cart",
    initialState:{
        products: [],
        quantity:0,
        total:0
    },
    reducers:{
        addProduct: (state, action) => {
            state.quantity += action.payload.quantity;
            state.products.push(action.payload)
            state.total += action.payload.price * action.payload.quantity;
        },
        deleteProduct: (state, action) => {
            state.products= [];
            state.quantity= 0;
            state.total= 0;
        },
        addOne: (state, action) => {
            state.quantity += 1;
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0) {
                state.products[index].quantity += 1;
            }
        },
        subtractOne: (state, action) => {
            const index = state.products.findIndex(
                (item) => item._id === action.payload.id
            );
            if (index >= 0 && state.products[index].quantity > 0) {
                state.products[index].quantity -= 1;
                state.quantity -= 1;
            }
        },
        removeOne:(state, action) => {
            const index = state.products.findIndex(
                    (item) => item._id === action.payload.id
                );
            if (index >= 0) {
                    state.quantity -= state.products[index].quantity;

                    state.products.splice(index, 1);
            } else {
                    console.warn(
                      `Cant remove product (id: ${action.payload.id}) as its not in basket!`
                    )
            }
        }
    }
})

export const {addProduct,  deleteProduct, addOne, subtractOne, removeOne} = cart.actions

export default cart.reducer;

export const emptyCart = async(dispatch) => {
    try{
      dispatch(deleteProduct());
    } catch(e){
      console.log(e)
    }
  }

