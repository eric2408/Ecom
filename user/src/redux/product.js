import { createSlice } from "@reduxjs/toolkit";
import { userRequest } from "../requestAxios";

export const productSlice = createSlice({
  name: "product",
  initialState: {
    products: [],
    isFetching: false,
    error: false,
  },
  reducers: {
    //Get all products
    getProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    getProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products = action.payload;
    },
    getProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Delete Products
    deleteProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    deleteProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.splice(state.products.findIndex((item) => item._id === action.payload), 1);
    },
    deleteProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Update products
    updateProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    updateProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products[
        state.products.findIndex((item) => item._id === action.payload.id)
      ] = action.payload.product;
    },
    updateProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },

    //Add Product
    addProductStart: (state) => {
      state.isFetching = true;
      state.error = false;
    },
    addProductSuccess: (state, action) => {
      state.isFetching = false;
      state.products.push(action.payload);
    },
    addProductFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
  },
});

export const {
  getProductStart,
  getProductSuccess,
  getProductFailure,
  deleteProductStart,
  deleteProductSuccess,
  deleteProductFailure,
  updateProductStart,
  updateProductSuccess,
  updateProductFailure,
  addProductStart,
  addProductSuccess,
  addProductFailure,
} = productSlice.actions;

export default productSlice.reducer;

export const getProducts = async (dispatch) => {
    dispatch(getProductStart());
    try {
      const res = await userRequest.get("/products");
      dispatch(getProductSuccess(res.data));
    } catch (err) {
      dispatch(getProductFailure());
    }
  };


export const deleteProduct = async (id, dispatch) => {
  dispatch(deleteProductStart());
  try {
    const res = await userRequest.delete(`/products/${id}`);
    dispatch(deleteProductSuccess(id));
  } catch (err) {
    dispatch(deleteProductFailure());
  }
};

export const updateProduct = async (id, product, dispatch) => {
  dispatch(updateProductStart());
  try {
    const res = await userRequest.put(`/products/${id}`, product);
    dispatch(updateProductSuccess({ id, product }));
  } catch (err) {
    dispatch(updateProductFailure());
  }
};
export const addProduct = async (product, dispatch) => {
  dispatch(addProductStart());
  try {
    const res = await userRequest.post(`/products`, product);
    dispatch(addProductSuccess(res.data));
  } catch (err) {
    dispatch(addProductFailure());
  }
};