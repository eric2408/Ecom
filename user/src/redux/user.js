import { createSlice } from "@reduxjs/toolkit";
import { generalRequest } from "../requestAxios";

const user = createSlice({
  name: "user",
  initialState: {
    currentUser: null,
    isFetching: false,
    error: false,
  },
  reducers: {
    loginStart: (state) => {
      state.isFetching = true;
    },
    loginSuccess: (state, action) => {
      state.isFetching = false;
      state.error = false;
      state.currentUser = action.payload;
    },
    loginFailure: (state) => {
      state.isFetching = false;
      state.error = true;
    },
    logOut: (state) => {
      state.currentUser = null;
    },
  },
});

export const { loginStart, loginSuccess, loginFailure, logOut } = user.actions;
export default user.reducer;


export const login = async(dispatch, user) =>{
    dispatch(loginStart());
    try{
        const res = await generalRequest.post('/auth/login', user)
        dispatch(loginSuccess(res.data))
    } catch(e){
        dispatch(loginFailure())
    }
}

export const logout = async(dispatch, user) => {
  try{
    dispatch(logOut());
  } catch(e){
    console.log(e)
  }
}

