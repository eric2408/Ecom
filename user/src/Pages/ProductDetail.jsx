import React, { useEffect, useState } from 'react'
import styled from "styled-components";
import Announcement from '../Components/Announcement';
import Footer from '../Components/Footer';
import Navbar from '../Components/Navbar';
import Subscription from '../Components/Subscription';
import RemoveIcon from '@mui/icons-material/Remove';
import AddIcon from '@mui/icons-material/Add';
import { mobileScreen } from '../Helper';
import { useLocation } from "react-router";
import {generalRequest} from '../requestAxios';
import { addProduct } from '../redux/cart';
import { useDispatch } from 'react-redux';

const Container = styled.div``;

const Padding = styled.div`
  padding: 50px;
  display: flex;
  ${mobileScreen({ padding: "10px", flexDirection:"column" })}
`;

const ImgContainer = styled.div`
  flex: 1;
`;

const Img = styled.img`
  width: 100%;
  height: 90vh;
  object-fit: cover;
  ${mobileScreen({ height: "40vh" })}
`;

const InfoContainer = styled.div`
  flex: 1;
  padding: 0px 50px;
  ${mobileScreen({ padding: "10px" })}
`;

const Title = styled.h1`
  font-weight: 200;
`;

const Description = styled.p`
  margin: 20px 0px;
`;

const Price = styled.span`
  font-weight: 100;
  font-size: 40px;
`;

const FilterContainer = styled.div`
  width: 50%;
  margin: 30px 0px;
  display: flex;
  justify-content: space-between;
  ${mobileScreen({ width: "100%" })}
`;

const Filter = styled.div`
  display: flex;
  align-items: center;
`;

const FilterTitle = styled.span`
  font-size: 20px;
  font-weight: 200;
`;

const ColorOptions = styled.div`
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background-color: ${(props) => props.color};
  margin: 0px 5px;
  cursor: pointer;
`;

const FilterSize = styled.select`
  margin-left: 10px;
  padding: 5px;
`;

const SizeOption = styled.option``;

const Quantity = styled.div`
  width: 50%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  ${mobileScreen({ width: "100%" })}
`;

const AmountContainer = styled.div`
  display: flex;
  align-items: center;
  font-weight: 700;
`;

const Amount = styled.span`
  width: 30px;
  height: 30px;
  border-radius: 10px;
  border: 1px solid teal;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0px 5px;
`;

const Button = styled.button`
  padding: 15px;
  border: 2px solid skyblue;
  background-color: white;
  cursor: pointer;
  font-weight: 500;
  &:hover {
    background-color: skyblue;
    color: white;
  }
`;

function ProductDetail() {
  const location = useLocation();
  const dispatch = useDispatch();

  const id = location.pathname.split('/')[2];
  const [product, setProduct] = useState({});
  const [quantity, setQuantity] = useState(1);
  const [color, setColor] = useState("");
  const [size, setSize] = useState("");

  useEffect(() => {
    const getProduct = async () => {
      try {
        const res = await generalRequest.get("/products/" + id);
        setProduct(res.data);
      } catch {}
    };
    getProduct();
  }, [id]);


  const handleClick = (sign) => {
    if(sign === 'subtract' ){
      quantity > 1 && setQuantity(quantity -1)
    } else {
      setQuantity(quantity +1)
    }
  }

  const handleAddCart =()=>{
    dispatch(addProduct({... product, quantity, color, size }))
  }

  return (
    <Container>
        <Announcement />
        <Navbar />
        <Padding>
            <ImgContainer>
                <Img src={product.img} />
            </ImgContainer>
            <InfoContainer>
                <Title>{product.title}</Title>
                <Description>
                  {product.description}
                </Description>
                <Price>$ {product.price}</Price>
                <FilterContainer>
                <Filter>
                    <FilterTitle>Color</FilterTitle>
                    {product.color?.map((c) => (
                      <ColorOptions color={c} key={c} onClick={()=> setColor(c)}/>
                    ))}
                </Filter>
                <Filter>
                    <FilterTitle>Size</FilterTitle>
                    <FilterSize onClick={(e)=> setSize(e.target.value)}>
                    {product.size?.map((s) => (
                    <SizeOption key={s}>{s}</SizeOption>
                ))}
                    </FilterSize>
                </Filter>
                </FilterContainer>
                <Quantity>
                    <AmountContainer>
                        <RemoveIcon onClick={()=> handleClick('subtract')}/>
                        <Amount>{quantity}</Amount>
                        <AddIcon onClick={()=> handleClick('addition')}/>
                    </AmountContainer>
                    <Button onClick={handleAddCart}>ADD TO CART</Button>
                </Quantity>
            </InfoContainer>
        </Padding>
        <Subscription />
        <Footer />
  </Container>
  )
}

export default ProductDetail