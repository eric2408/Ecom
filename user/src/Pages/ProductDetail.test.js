import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import {store, persistor} from '../redux/store';
import{Provider} from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import ProductDetail from './ProductDetail'

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <BrowserRouter>
        <ProductDetail />
      </BrowserRouter>
    </PersistGate>
  </Provider>

  );
  expect(asFragment()).toMatchSnapshot();
});