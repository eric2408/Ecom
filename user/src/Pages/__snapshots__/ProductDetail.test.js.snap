// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jIRcFI"
  >
    <div
      class="sc-bcXHqe eLQayE"
    >
      Free Shipping on Orders Over $49!
    </div>
    <div
      class="sc-eDWCr lhYMNP"
    >
      <div
        class="sc-bqWxrE gvnzob"
      >
        <div
          class="sc-ksBlkl cxUMGU"
        >
          <span
            class="sc-hBxehG ddBtzW"
          >
            EN
          </span>
          <div
            class="sc-fnGiBr dKybLJ"
          >
            <input
              class="sc-fEXmlR jPVLHP"
              placeholder="search"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              style="color: white; margin-left: 8px; font-size: 20px; cursor: pointer;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-bjfHbI kLHHpZ"
        >
          <a
            href="/"
            style="text-decoration: none;"
          >
            <h1
              class="sc-idXgbr csUYOV"
            >
              VIRTUAL SPACE
            </h1>
          </a>
        </div>
        <div
          class="sc-dIfARi ckAuwl"
        >
          <a
            href="/register"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-dmctIk hHlIX"
            >
              <div
                class="sc-hHTYSt giXObl"
              >
                REGISTER
              </div>
               
            </div>
          </a>
          <a
            href="/login"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hHTYSt giXObl"
            >
              SIGN IN
            </div>
          </a>
          <a
            href="/cart"
            style="text-decoration: none; color: white;"
          >
            <div
              class="sc-hHTYSt giXObl"
            >
              <span
                class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ShoppingCartIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge BaseBadge-invisible css-fu0map-MuiBadge-badge"
                />
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-hhOBVt lfWJpK"
    >
      <div
        class="sc-gYbzsP iIWarg"
      >
        <img
          class="sc-cCjUiG kPSDBh"
        />
      </div>
      <div
        class="sc-cjibBx xBFQH"
      >
        <h1
          class="sc-jcMfQk jNzXtb"
        />
        <p
          class="sc-cabOPr heUypS"
        />
        <span
          class="sc-iTFTee dtbDsn"
        >
          $ 
        </span>
        <div
          class="sc-iAEawV hDbrGB"
        >
          <div
            class="sc-eeMvmM jXVIBn"
          >
            <span
              class="sc-cUEOzv evzqlJ"
            >
              Color
            </span>
          </div>
          <div
            class="sc-eeMvmM jXVIBn"
          >
            <span
              class="sc-cUEOzv evzqlJ"
            >
              Size
            </span>
            <select
              class="sc-dPWrhe gcDbKJ"
            />
          </div>
        </div>
        <div
          class="sc-cOxWqc jvlXEp"
        >
          <div
            class="sc-AHaJN zLkQu"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
            <span
              class="sc-khsqcC bpCGuK"
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </div>
          <button
            class="sc-hTBuwn jvmUfc"
          >
            ADD TO CART
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-lllmON OINtR"
    >
      <div
        class="sc-cwSeag cnCIrx"
      >
        <div
          class="sc-iJnaPW eICDBr"
        >
          <h2
            class="sc-gikAfH cYPRpy"
          >
            Our Advantages
          </h2>
        </div>
        <div
          class="sc-ezOQGI hlpfXw"
        >
          <div
            class="sc-bYMpWt bZFLug"
          >
            <div
              class="sc-kMjNwy dSRJqQ"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H112C85.5 0 64 21.5 64 48v48H8c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h272c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H40c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h208c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H8c-4.4 0-8 3.6-8 8v16c0 4.4 3.6 8 8 8h208c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H64v128c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
                />
              </svg>
            </div>
            <div
              class="sc-ilhmMj kBkqbx"
            >
              <h4>
                Express Delivery
              </h4>
              <p>
                Ships within 24 Hours
              </p>
            </div>
          </div>
          <div
            class="sc-bYMpWt bZFLug"
          >
            <div
              class="sc-kMjNwy dSRJqQ"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M466.5 83.7l-192-80a48.15 48.15 0 0 0-36.9 0l-192 80C27.7 91.1 16 108.6 16 128c0 198.5 114.5 335.7 221.5 380.3 11.8 4.9 25.1 4.9 36.9 0C360.1 472.6 496 349.3 496 128c0-19.4-11.7-36.9-29.5-44.3zM256.1 446.3l-.1-381 175.9 73.3c-3.3 151.4-82.1 261.1-175.8 307.7z"
                />
              </svg>
            </div>
            <div
              class="sc-ilhmMj kBkqbx"
            >
              <h4>
                Safe Purchases
              </h4>
              <p>
                Full Refund
              </p>
            </div>
          </div>
          <div
            class="sc-bYMpWt bZFLug"
          >
            <div
              class="sc-kMjNwy dSRJqQ"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                />
              </svg>
            </div>
            <div
              class="sc-ilhmMj kBkqbx"
            >
              <h4>
                Exciting Deals
              </h4>
              <p>
                On all prepaid orders
              </p>
            </div>
          </div>
          <div
            class="sc-bYMpWt bZFLug"
          >
            <div
              class="sc-kMjNwy dSRJqQ"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 432c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V256H0v176zm192-68c0-6.6 5.4-12 12-12h136c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-40zm-128 0c0-6.6 5.4-12 12-12h72c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM576 80v48H0V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48z"
                />
              </svg>
            </div>
            <div
              class="sc-ilhmMj kBkqbx"
            >
              <h4>
                Secure Payments
              </h4>
              <p>
                Through Stripe and Paypal
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-gswNZR jofBsk"
    >
      <div
        class="sc-dkrFOg kLzEQW"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <h1
            class="sc-hLBbgP cfrXqI"
          >
            VIRTUAL SPACE
          </h1>
        </a>
        <p
          class="sc-eDvSVe jvdQks"
        >
          Join Virtual Space! Time go a fascinating world!
        </p>
        <div
          class="sc-jSUZER bHEyKf"
        >
          <div
            class="sc-gKPRtg bThFsx"
            color="4267B2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FacebookIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
              />
            </svg>
          </div>
          <div
            class="sc-gKPRtg eqicWh"
            color="cd486b"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="InstagramIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
          </div>
          <div
            class="sc-gKPRtg fQhJD"
            color="1DA1F2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="TwitterIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-iBYQkv jBFHGg"
      >
        <h3
          class="sc-ftTHYK gFQvOI"
        >
          Support
        </h3>
        <ul
          class="sc-pyfCe bBXOyA"
        >
          <li
            class="sc-jrcTuL fBSxcv"
          >
            About Virtual Space
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Account
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Orders
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Favorites
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            My Cart
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Privacy Policy
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Refund Policy
          </li>
          <li
            class="sc-jrcTuL fBSxcv"
          >
            Terms of Service
          </li>
        </ul>
      </div>
      <div
        class="sc-kDvujY hcROd"
      >
        <h3
          class="sc-ftTHYK gFQvOI"
        >
          Contact Us
        </h3>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           27 San Carlos St. Buffalo, NY 14215
        </div>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PhoneIphoneIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 1h-8C6.12 1 5 2.12 5 3.5v17C5 21.88 6.12 23 7.5 23h8c1.38 0 2.5-1.12 2.5-2.5v-17C18 2.12 16.88 1 15.5 1zm-4 21c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4.5-4H7V4h9v14z"
            />
          </svg>
           +1 234 56 7899
        </div>
        <div
          class="sc-ipEyDJ hXqFPG"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            style="margin-right: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
           jo123@gmail.com
        </div>
        <img
          class="sc-csuSiG dabsRR"
          src="https://chargie.org/wp-content/uploads/2021/03/PayPal-stripe-web-opt-300x136-1-300x136-300x136-1.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
